dashboard/15-final/app/login/page.tsx

```diff
@@ -1,6 +1,13 @@
 import LoginForm from '@/app/ui/login-form';
+import { getServerSession } from 'next-auth/next';
+import { redirect } from 'next/navigation';
+import { authOptions } from '@/app/api/auth/[...nextauth]/route';
+
+export default async function Page() {
+  // Get user session token
+  const session = await getServerSession(authOptions);
+  if (session) redirect('/dashboard');
```

Comment:
Redirects should be done on the Middleware-level, this will avoid even starting to render this page.